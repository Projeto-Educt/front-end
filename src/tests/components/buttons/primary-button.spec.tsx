import PrimaryButton from '@/components/button/primary-button';
import { fireEvent, render, screen } from '@testing-library/react';

describe('PrimaryButton', () => {
  it('should render correctly', () => {
    render(<PrimaryButton>Jest</PrimaryButton>);

    const button = screen.getByRole('button', { name: 'Jest' });

    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('type', 'button');
  });

  it('should render correctly props', () => {
    render(<PrimaryButton type="submit">Jest</PrimaryButton>);

    const button = screen.getByRole('button', { name: 'Jest' });

    expect(button).toHaveAttribute('type', 'submit');
  });

  it('Should execute onClick', () => {
    const onClick = jest.fn();
    render(<PrimaryButton onClick={onClick}>Jest</PrimaryButton>);

    const button = screen.getByRole('button', { name: 'Jest' });

    expect(onClick).not.toHaveBeenCalled();
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('Should match snapshot', () => {
    const { container } = render(<PrimaryButton>Jest</PrimaryButton>);

    expect(container).toMatchSnapshot();
  });
});
