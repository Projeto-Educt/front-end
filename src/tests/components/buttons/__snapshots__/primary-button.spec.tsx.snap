// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton Should match snapshot 1`] = `
<div>
  <button
    class="text-white bg-primary-600 rounded-md h-10 text-center w-full font-bold undefined"
    type="button"
  >
    Jest
  </button>
</div>
`;
